node/src/Pos/Wallet/Web/Pending/Types.hs

```diff
@@ -0,0 +1,62 @@
+-- | Types related to pending transactions.
+
+module Pos.Wallet.Web.Pending.Types
+    ( PendingTx (..)
+    , ptxTxId
+    , ptxTxAux
+    , ptxCond
+    , ptxCreationSlot
+    , ptxWallet
+
+    , PtxCondition (..)
+    , PtxBlockInfo
+    ) where
+
+import           Universum
+
+import           Control.Lens                     (makeLenses)
+import           Pos.Core.Types                   (SlotId)
+import           Pos.Txp.Core.Types               (TxAux, TxId)
+import           Pos.Wallet.Web.ClientTypes.Types (CId, Wal)
+
+-- | Required information about block where given pending transaction is sited
+type PtxBlockInfo = SlotId
+
+-- | Current state of pending transaction.
+--
+-- Once transaction is created, it should be assigned 'PtxApplying' condition
+-- in order to be tracked by resubmitter.
+--
+-- When wallet tracker notices a block with given transaction, it switches
+-- condition to 'PtxInUpperBlocks' providing needed information about that block.
```

Comment:
why `Upper`? I couldn't get it, got eventually. IHMO it should be named `Newest` AFAIK

also `PtxApplying` doesn't reflect it's not in the blockchain. but it's subjective. I don't mind to use this name.